packages/compat/package.json

```diff
@@ -55,6 +55,14 @@
     "rollup": "^4.16.2",
     "typescript": "^5.4.5"
   },
+  "peerDependencies": {
+    "@types/eslint": "^9.6.0"
```

Comment:
Now that the `eslint` package itself publishes its own types (https://github.com/eslint/eslint/pull/18854), `@types/eslint` is no longer necessary in general. I'd assumed the plan was to turn it into a stub type definition per normal DefinitelyTyped practices. Should this be switched to `"eslint"`, along with the meta lower on?

Reply:
> I'd assumed the plan was to turn it into a stub type definition per normal DefinitelyTyped practices.

We don't know normal DefiniteTyped practices. 😄 Can you open an issue somewhere for this task explaining what it is?

Reply:
#18928

Reply:
Great, per https://github.com/eslint/eslint/issues/18928#issuecomment-2364996476 I'll mark this as needing to switch from `@types/eslint` to `eslint`. Thanks @snitin315!

Reply:
switched to `eslint ^9.10.0` (first version with bundled typings)
@nzakas @JoshuaKGoldberg Question: Should the "optional" be removed? (Can this package be used without eslint?)

Reply:
I think optional can stay. This package doesn't use `eslint` at runtime, only at dev time to validate types. @JoshuaKGoldberg?

Reply:
👍